server/src/test/java/org/elasticsearch/snapshots/RepositoriesIT.java

```diff
@@ -225,4 +231,59 @@ public void testRepositoryVerification() {
             assertThat(ExceptionsHelper.stackTrace(ex), containsString("is not shared"));
         }
     }
+
+    public void testRepositoryConflict() throws Exception {
+        logger.info("--> creating repository");
+        final String repo = "test-repo";
+        assertAcked(client().admin().cluster().preparePutRepository(repo).setType("mock").setSettings(
+            Settings.builder()
+                .put("location", randomRepoPath())
+                .put("random", randomAlphaOfLength(10))
+                .put("wait_after_unblock", 200)).get());
+
+        logger.info("--> snapshot");
+        final String index = "test-idx";
+        assertAcked(prepareCreate(index, 1, Settings.builder().put("number_of_shards", 1).put("number_of_replicas", 0)));
+        for (int i = 0; i < 10; i++) {
+            indexDoc(index, Integer.toString(i), "foo", "bar" + i);
+        }
+        refresh();
+        final String snapshot1 = "test-snap1";
+        client().admin().cluster().prepareCreateSnapshot(repo, snapshot1).setWaitForCompletion(true).get();
+
+        String blockedNode = internalCluster().getMasterName();
+        ((MockRepository)internalCluster().getInstance(RepositoriesService.class, blockedNode).repository(repo)).blockOnDataFiles(true);
+        logger.info("--> start deletion of snapshot");
+        ActionFuture<AcknowledgedResponse> future = client().admin().cluster().prepareDeleteSnapshot(repo, snapshot1).execute();
+        logger.info("--> waiting for block to kick in on node [{}]", blockedNode);
+        waitForBlock(blockedNode, repo, TimeValue.timeValueSeconds(10));
+
+        logger.info("--> try deleting the repository, should fail because the deletion of the snapshot is in progress");
+        try {
```

Comment:
We actually have a neat short-cut for testing this kind of spot with the `expectThrows` method that keeps this much shorter. See https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java#L2169 for an example.